extensions/amp-story/1.0/amp-story.js

```diff
@@ -604,6 +626,55 @@ export class AmpStory extends AMP.BaseElement {
     this.installGestureRecognizers_();
   }
 
+  /**
```

Comment:
I don't love putting +50 lines of helper utils in amp-story, when they're all related to each other.  We don't want amp-story's filesize to grow O(n) for n features 😅

Maybe add an `expanded-overlay.js` or similar?  WDYT @Enriqe @gmajoulet?